src/apps/replication/lib/prepare_list.cpp

```diff
@@ -122,32 +127,20 @@ error_code prepare_list::prepare(mutation_ptr& mu, partition_status status)
     //    return err;
      
     case PS_INACTIVE: // only possible during init  
-        err = ERR_OK;
         if (mu->data.header.last_committed_decree > max_decree())
         {
             reset(mu->data.header.last_committed_decree);
         }
         else if (mu->data.header.last_committed_decree > _last_committed_decree)
         {
-            for (decree d = last_committed_decree() + 1; d <= mu->data.header.last_committed_decree; d++)
+            // all mutations with lower decree must be ready
+            commit(mu->data.header.last_committed_decree, COMMIT_TO_DECREE_HARD);
+            // pop committed mutations if buffer is full
+            while (d - min_decree() >= capacity() && last_committed_decree() > min_decree())
```

Comment:
on really, the commit() may move forward the last_committed_decree(), then we can pop mutations to make the prepare list not full.